runtime/ms_rest/lib/ms_rest/http_operation_request.rb

```diff
@@ -77,14 +77,22 @@ def run_promise(&block)
           end
         end
 
-        @connection.run_request(:"#{method}", build_path, body, {'User-Agent' => user_agent}.merge(headers)) do |req|
-          req.params = req.params.merge(query_params.reject{|_, v| v.nil?}) unless query_params.nil?
-          yield(req) if block_given?
+        loop do
+          @response = @connection.run_request(:"#{method}", build_path, body, {'User-Agent' => user_agent}.merge(headers)) do |req|
+            req.params = req.params.merge(query_params.reject{|_, v| v.nil?}) unless query_params.nil?
+            yield(req) if block_given?
+          end
+
+          break if ((@response.status != 429) || (@response.status == 429 && @response.headers['retry-after'].nil?))
+
+          if(@response.status == 429 && !@response.headers['retry-after'].nil?)
```

Comment:
is this retry behavior general or Azure only? should it belong to ms_rest or ms_rest_azure?

Reply:
The throttling seems to be general in all other SDKs. So, following the same.